File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/analysis/QueryAnalysisTestSuite.scala
package dev.typr.foundationssc.docs.analysis
import dev.typr.foundationssc.*

@SuppressWarnings(Array("unused"))
object QueryAnalysisTestSuite:
  private val transactor: Transactor = null // placeholder

  //start
  def allQueriesTypeCheck(): Unit =
    val analyzables = AnalyzableScanner.scan("com.myapp.db")
    val checker = QueryChecker.create(transactor)
    checker.checkAll(analyzables)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/analysis/QueryAnalysisTestSuite.java
package dev.typr.foundations.docs.analysis;

import dev.typr.foundations.AnalyzableScanner;
import dev.typr.foundations.QueryChecker;
import dev.typr.foundations.Transactor;

@SuppressWarnings("unused")
public class QueryAnalysisTestSuite {
    private final Transactor transactor = null; // placeholder

    //start
    void allQueriesTypeCheck() {
        var analyzables = AnalyzableScanner.scan("com.myapp.db");
        QueryChecker checker = () -> transactor;
        checker.checkAll(analyzables);
    }
    //stop
}
